### PR TITLE
update LastTransitionTime for pod ready condition when node notready

### PR DESCRIPTION
**Pull request by Pingan2017**
…tready

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/sig node
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/util/node/controller_utils.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (mattjmcnaughton) [inline pkg/controller/util/node/controller_utils.go]: Nit: I personally don't care, but I think some people have opinions on the `v1 ".../v1"` vs `".../v1` debate - was this change intentional/do you know if k8s has an overall policy?

File: pkg/controller/util/node/controller_utils.go [hunk 2]
```diff
@@ -135,8 +135,9 @@ func MarkAllPodsNotReady(kubeClient clientset.Interface, node *v1.Node) error {
 		}
 
 		for i, cond := range pod.Status.Conditions {
-			if cond.Type == v1.PodReady {
+			if cond.Type == v1.PodReady && cond.Status != v1.ConditionFalse {
 				pod.Status.Conditions[i].Status = v1.ConditionFalse
+				pod.Status.Conditions[i].LastTransitionTime = metav1.Now()
```
Comment (mattjmcnaughton) [inline pkg/controller/util/node/controller_utils.go]: Could we replace this with `pkg/api/v1/pod/util.go#UpdatePodCondition`? I think it includes all of the updates we want (i.e. updating `LastTransitionTime`) and has the advantage of centralizing the logic for updating pod condition in a central place.

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
Love the idea :)

Small request from me re utilizing a utility function so that we can keep the functionality for updating pod conditions in a central place.

File: pkg/controller/util/node/controller_utils.go (hunk 1 above)
Comment (Pingan2017) [reply pkg/controller/util/node/controller_utils.go]: it's changed by vscode automatically :smile:

File: pkg/controller/util/node/controller_utils.go (hunk 2 above)
Comment (Pingan2017) [reply pkg/controller/util/node/controller_utils.go]: updated!

File: pkg/controller/util/node/controller_utils.go [hunk 3]
```diff
@@ -134,9 +135,12 @@ func MarkAllPodsNotReady(kubeClient clientset.Interface, node *v1.Node) error {
 			continue
 		}
 
-		for i, cond := range pod.Status.Conditions {
+		for _, cond := range pod.Status.Conditions {
 			if cond.Type == v1.PodReady {
-				pod.Status.Conditions[i].Status = v1.ConditionFalse
+				cond.Status = v1.ConditionFalse
+				if !utilpod.UpdatePodCondition(&pod.Status, &cond) {
+					break
```
Comment (mattjmcnaughton) [inline pkg/controller/util/node/controller_utils.go]: Just confirming that we can be confident that `pod.Status.Conditions` will only contain one `condition` for which `Type = v1.PodReady`? Or alternatively, if there are multiple, we only want to examine one of them?


Confirming my understanding... we can break here because if `UpdatePodCondition` returns false, it means no update is necessary for this condition... and we can be confident that `pod.Status.Condition

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Would love someone slightly more familiar with the code base than me to take a look at the question I asked, but other than that, looks good :)

File: pkg/controller/util/node/controller_utils.go (hunk 3 above)
Comment (Pingan2017) [reply pkg/controller/util/node/controller_utils.go]: yeah, only one PodReady condition

**Comment by Pingan2017** [comment]
/assign @deads2k
ping @deads2k @liggitt for review
